protocols/gossipsub/src/behaviour.rs

```diff
@@ -1031,7 +1036,9 @@ impl Gossipsub {
 
     /// Takes each control action mapping and turns it into a message
     fn flush_control_pool(&mut self) {
-        for (peer, controls) in self.control_pool.drain().collect::<Vec<_>>() {
+        let mut old = BTreeMap::new();
+        std::mem::swap(&mut self.control_pool, &mut old);
+        for (peer, controls) in old.into_iter() {
```

Comment:
This is necessary because BTreeMap does not have drain(), so basically manual drain.